File: app/src/main/java/com/ypg/neville/model/utils/ColorPickerManager.java
package com.ypg.neville.model.utils;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AlertDialog;
import androidx.preference.PreferenceManager;

import com.skydoves.colorpickerview.ColorPickerView;
import com.skydoves.colorpickerview.listeners.ColorListener;
import com.skydoves.colorpickerview.sliders.AlphaSlideBar;
import com.skydoves.colorpickerview.sliders.BrightnessSlideBar;
import com.ypg.neville.MainActivity;
import com.ypg.neville.R;


public class ColorPickerManager {

    public static  int Color;


    /**
     * Muestra un cuadro de selección de colores
     * @param context
     * @param color Color por defecto que será seleccionado
     * @param key_Preferences key en el archivo de preferencia donde será almacenado el color
     */
    public static void  ShowColorPicker(Context context,  int color, String key_Preferences, String title){

        AlertDialog.Builder alert = new AlertDialog.Builder(context);
        alert.setTitle(title);

        View view = LayoutInflater.from(context).inflate(R.layout.color_picker,null);
        ColorPickerView colorPickerView = view.findViewById(R.id.color_picker);
        AlphaSlideBar alphaSlideBar = view.findViewById(R.id.alphaSlideBar);
        BrightnessSlideBar brightnessSlideBar = view.findViewById(R.id.brightnessSlide);

        Button button = view.findViewById(R.id.colorpicker_btn_ok);

        colorPickerView.attachAlphaSlider(alphaSlideBar);
        colorPickerView.attachBrightnessSlider(brightnessSlideBar);



        //estableciendo un color inicial
        if (color != 0 ){
            colorPickerView.setInitialColorRes(color);
        }

        alert.setView(view);
        AlertDialog alertDialog = alert.create();

        colorPickerView.setColorListener(new ColorListener() {
            @Override
            public void onColorSelected(int color, boolean fromUser) {
                    button.setBackgroundColor(color);
                    ColorPickerManager.Color = color;
            }
        });
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                PreferenceManager.getDefaultSharedPreferences(context).edit().putInt(key_Preferences, ColorPickerManager.Color).apply();

                if(MainActivity.mainActivityThis != null){
                    MainActivity.mainActivityThis.AuxSetColorBar(ColorPickerManager.Color);
                }
                alertDialog.dismiss();
            }
        });


        alertDialog.show();

    }

}

File: app/src/main/java/com/ypg/neville/model/utils/ColorPickerManager.kt
package com.ypg.neville.model.utils

import android.content.Context
import androidx.appcompat.app.AlertDialog
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.Slider
import androidx.compose.material3.Text
import androidx.compose.runtime.mutableFloatStateOf
import androidx.compose.runtime.mutableIntStateOf
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.ComposeView
import androidx.compose.ui.unit.dp
import androidx.core.content.edit
import com.ypg.neville.model.preferences.DbPreferences
import com.ypg.neville.MainActivity

object ColorPickerManager {

    @JvmField
    var colorValue: Int = 0

    @JvmStatic
    fun showColorPicker(context: Context, color: Int, keyPreferences: String, title: String) {
        val compose = ComposeView(context)
        val alertDialog = AlertDialog.Builder(context)
            .setTitle(title)
            .setView(compose)
            .create()

        compose.setContent {
            com.ypg.neville.ui.theme.NevilleTheme {
                val initialColor = if (color != 0) color else 0xFF5A6270.toInt()
                val initial = Color(initialColor)
                val red = remember { mutableFloatStateOf(initial.red * 255f) }
                val green = remember { mutableFloatStateOf(initial.green * 255f) }
                val blue = remember { mutableFloatStateOf(initial.blue * 255f) }
                val selected = remember { mutableIntStateOf(initialColor) }

                fun recomputeColor() {
                    val r = red.floatValue.toInt().coerceIn(0, 255)
                    val g = green.floatValue.toInt().coerceIn(0, 255)
                    val b = blue.floatValue.toInt().coerceIn(0, 255)
                    selected.intValue = android.graphics.Color.rgb(r, g, b)
                    colorValue = selected.intValue
                }

                Column(
                    verticalArrangement = Arrangement.spacedBy(12.dp),
                    modifier = Modifier.padding(12.dp)
                ) {
                    Box(
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(68.dp)
                            .background(Color(selected.intValue), RoundedCornerShape(10.dp))
                    )

                    ChannelSlider("Rojo", red.floatValue) {
                        red.floatValue = it
                        recomputeColor()
                    }
                    ChannelSlider("Verde", green.floatValue) {
                        green.floatValue = it
                        recomputeColor()
                    }
                    ChannelSlider("Azul", blue.floatValue) {
                        blue.floatValue = it
                        recomputeColor()
                    }

                    Button(onClick = {
                        DbPreferences.default(context).edit {
                            putInt(keyPreferences, selected.intValue)
                        }
                        if (keyPreferences == "color_marcos") {
                            MainActivity.currentInstance()?.auxSetColorBar(selected.intValue)
                        }
                        alertDialog.dismiss()
                    }) {
                        Text("Aplicar")
                    }
                }
            }
        }

        alertDialog.show()
    }

    @androidx.compose.runtime.Composable
    private fun ChannelSlider(
        name: String,
        value: Float,
        onValueChange: (Float) -> Unit
    ) {
        Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                Text(name)
                Text(value.toInt().toString())
            }
            Slider(
                value = value,
                onValueChange = onValueChange,
                valueRange = 0f..255f
            )
        }
    }
}
